AWSAppSyncClient/Internal/SubscriptionWithSyncConfiguration.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+/// Configuration values for an AppSyncSubscriptionWithSync instance.
+public struct SubscriptionWithSyncConfiguration {
```

Comment:
I think we should remove the word `Subscription` from this struct name. We don't necessarily use subscription for the sync API. Maybe something like `SyncIntervalConfiguration`?

Reply:
I think `SyncIntervalConfiguration` can be used not only for sync interval parameter, so from my opinion `SyncConfiguration` is the best.

Reply:
> I think `SyncIntervalConfiguration` can be used not only for sync interval parameter, so from my opinion `SyncConfiguration` is the best.

That's my concern as well--if we're going to the bother of having a configuration type in the first place, it should probably be a bit broader than just an interval configuration.

Reply:
Resolved as noted elsewhere: `SyncConfiguration` is now a `struct`, and changed the initializer parameter and default pattern.